crates/motoko/src/lib/value.rs

```diff
@@ -128,6 +128,32 @@ pub enum Value {
     Collection(Collection),
     Dynamic(DynamicValue),
     // DynamicRef(DynamicRef),
+    Actor(Actor),
+    ActorMethod(ActorMethod),
+}
+
+/// Actor value.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Actor {
+    /// Provides the public interface for message sends (for sanity checks, warnings).
+    pub def: Option<ActorDef>,
+    /// Provides the operational target of message sends.
+    pub id: ActorId,
+}
+
+/// Actor identifier (key part of an actor value).
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum ActorId {
+    /// Actor is identified by a local name in the Agent program that creates it.
+    Local(Id),
+    // to do -- case: canister ID and canister alias pair from dfx.json.
```

Comment:
@rvanasa `ActorId` could have a variant called `DfxCanister` or whatever name seems appropriate given what identifier you choose to use.

As you've suggested previously, perhaps it's simplest to just use the "canister alias" dfx concept that we've discussed many times now as the identifier for the `DfxCanister` case of `ActorId`? (and eschew having the replica-created canister id as part of that id, e.g., as a pair.)

If so, perhaps we can instead put that extra ID info into the actor as associated data, when it's available.